src/Token/TokenBase.tsx

```diff
@@ -0,0 +1,92 @@
+import {KeyboardEvent} from 'react'
+import styled, {css} from 'styled-components'
+import {variant} from 'styled-system'
+import {get} from '../constants'
+
+export type TokenSizeKeys = 'sm' | 'md' | 'lg' | 'xl'
+
+export const tokenSizes: Record<TokenSizeKeys, number> = {
+  sm: 16,
+  md: 20,
+  lg: 24,
+  xl: 32
+}
+
+export const defaultTokenSize = 'md'
+
+export interface TokenBaseProps
+  extends Omit<React.HTMLProps<HTMLSpanElement | HTMLButtonElement | HTMLAnchorElement>, 'size' | 'id'> {
+  as?: 'button' | 'a' | 'span'
+  handleRemove?: () => void
+  isSelected?: boolean
+  tabIndex?: number
+  text: string
+  id?: number | string
+  variant?: TokenSizeKeys
```

Comment:
IMO we should only use the prop name "variant" when no other name would be more clear. In this case, `size` feels more clear to me:

```suggestion
  size?: TokenSize
```

Reply:
I think I set this to `variant` because of something to do with `styled-components`. I also prefer `size`, so I'll try changing it.

Reply:
This is possible: https://styled-system.com/variants/#custom-prop-name